modules/programs/navi.nix

```
@@ -0,0 +1,85 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+
+  cfg = config.programs.navi;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.navi = {
+    enable = mkEnableOption "Navi";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.navi;
+      defaultText = literalExpression "pkgs.navi";
+      description = "The package to use for the navi binary.";
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      defaultText = literalExpression "{ }";
+      example = literalExpression ''
+        {
+          cheats = {
+            paths = [
+              "~/cheats/"
+            ];
+          };
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/navi/config.yaml</filename> on Linux
+        or <filename>~/Library/Application Support/navi/config.yaml</filename> on Darwin. See
```

I think the config directory can be generalized

```suggestion
        <filename>$XDG_CONFIG_HOME/navi/config.yaml</filename> on Linux
        or <filename>$HOME/Library/Application Support/navi/config.yaml</filename> on Darwin. See
```